files-community/Files
# Code Quality: Create alike RichCommands inheritance

**Resolved / Related Issues**
   Closes #13132

[x] Run as commands
[x] Compress into archive commands
[x] Decompress archive commands
[ ] Restore recycle bin commands (I dont' think inheritance is necessary here. The two classes share only a few lines of code)
[x] Tab closing commands

**Validation**
How did you test these changes?
   1. Run As
       -  Right click on an executable file use `Run as admin` and `Run as another user`
   2. Compress into archive
       - Right click on some files/folders use the `Create archive` commands
   3. Decompress archive
       - Select an archive or navigate into it and use all the `Decompress archive` commands
    4. Close tab
       - Open a few tab, right click on one of them and try one command
       - Repeat for other commands
